src/app/shared/directives/autofocus/directive.ts

```diff
@@ -26,16 +26,21 @@ export class AutofocusDirective implements AfterViewInit, OnDestroy {
 
   constructor(private readonly _el: ElementRef) {}
 
+  focus() {
+    setTimeout(() => this._el.nativeElement.focus());
+  }
+
   ngAfterViewInit(): void {
-    if (!this.opened) {
-      throw new Error('[opened] event binding is undefined');
+    if (this.opened) {
+      this.opened
+        .pipe(takeUntil(this._unsubscribe))
+        .pipe(filter(open => open))
+        .subscribe(_ => {
```

Comment:
nit: You can skip {} if there is just one line inside.

Reply:
I'd actually pass the focus method directly as subscribe parameter.
```typescript
...
.subscribe(this._focus.bind(this))
```